locals.tf

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#--------#
+# Locals #
+#--------#
+locals {
+  random_hash             = "${random_id.random_project_id_suffix.hex}"
+  root_resource_id        = "${var.org_id != "" ? "organizations/${var.org_id}" : var.folder_id != "" ? "folders/${var.folder_id}" : ""}"
+  vpc_host_project_id     = "${var.vpc_host_project_id != "" ? var.vpc_host_project_id : var.project_id}"
```

Comment:
This is confusing; why would we assign the value of `var.project_id` to `vpc_host_project_id`? If we don't have a host project we shouldn't provide an alternate value, because this tripped me up when looking at where the client VM was created.